src/widgets/barchart/bar.rs

```diff
@@ -65,8 +65,8 @@ impl<'a> Bar<'a> {
     /// display the label **in** the bar.
     /// See [`BarChart::direction`](crate::widgets::BarChart::direction) to set the direction.
     #[must_use = "method moves the value of self and returns the modified value"]
-    pub fn label(mut self, label: Line<'a>) -> Self {
-        self.label = Some(label);
+    pub fn label(mut self, label: impl Into<Line<'a>>) -> Self {
```

Comment:
Generally, to assist newer rustaceans, we add docs that call out that the method accepts the types that can be converted into line. The rationale behind this was seeing many newer dev's code doing things like `foo.value(Line::from(Span::from("bar")))` instead of the much simpler `foo.value("bar")`, and a desire to help users use the library effectively.

Example: https://docs.rs/ratatui/latest/ratatui/widgets/struct.Tabs.html#impl-Tabs%3C'a%3E

>titles can be a [Vec](https://doc.rust-lang.org/nightly/alloc/vec/struct.Vec.html) of [&str](https://doc.rust-lang.org/nightly/std/primitive.str.html), [String](https://doc.rust-lang.org/nightly/alloc/string/struct.String.html) or anything that can be converted into [Line](https://docs.rs/ratatui/latest/ratatui/text/struct.Line.html). As such, titles can be styled independently.

I'd suggest adding 3 examples (no asserts):
- `chart.label("str")`
- `chat.label(String::from("string"))`
- `chat.label(Line::from("line").red())`

Same idea applies to the other changes.